modules/parent-join/src/main/java/org/elasticsearch/join/query/HasChildQueryBuilder.java

```diff
@@ -460,6 +460,10 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws I
     @Override
     protected void extractInnerHitBuilders(Map<String, InnerHitContextBuilder> innerHits) {
         if (innerHitBuilder != null) {
+            if (innerHits.containsKey(innerHitBuilder.getName())) {
+                throw new IllegalArgumentException("innerHits already contains an entry for key [" + innerHitBuilder.getName() + "]");
```

Comment:
we should use the same format for the message than the option in the request, can you change `innerHits` to `[inner_hits]`  ?